src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java

```diff
@@ -156,8 +156,10 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
         Result previousBuildResult = null;
         AbstractBuild<?, ?> previousBuild = build.getPreviousCompletedBuild();
 
-        if (forceNewIssue && currentBuildResult == Result.FAILURE) {
-            currentBuildResultFailure(build, listener, null, filename, vars);
+        if (forceNewIssue) {
+            if (currentBuildResult == Result.FAILURE) {
+                currentBuildResultFailure(build, listener, null, filename, vars);
+            }
         }
```

Comment:
Yes thats true. But it also means if either of them is false, it goes to else. I don't want it to go to else if `forceNewIssue` is on. So if `currentBuildResult` is Result.SUCCESS, we do nothing and return true for `perform`. Before the change, we would go to else when `currentBuildResult` == `Result.SUCCESS` not caring about `forceNewIssue`. That is a behaviour we don't want